projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -2448,22 +2447,23 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
             });
         });
 
-        if (this.allowFiltering && this.filterMode === FilterMode.excelStyleFilter) {
-            this.closeExcelStyleDialog();
-        }
+        this.hideOverlays();
     }
 
-    private closeExcelStyleDialog() {
-        const excelStyleMenu = this.outlet.nativeElement.getElementsByClassName('igx-excel-filter__menu')[0];
-        if (excelStyleMenu) {
-            const overlay = this.overlayService.getOverlayById(excelStyleMenu.getAttribute('id'));
-            if (overlay) {
-                const animation = overlay.settings.positionStrategy.settings.closeAnimation;
-                overlay.settings.positionStrategy.settings.closeAnimation = null;
-                this.overlayService.hide(overlay.id);
-                overlay.settings.positionStrategy.settings.closeAnimation = animation;
-            }
-        }
+    /**
+    * @hidden
+    * @internal
+    */
+    public hideOverlays() {
+        this.overlayIDs.forEach(overlayID => {
+            this.overlayService.hide(overlayID);
+            // blur in case some editor somewhere decides to move focus back
+            this.overlayService.onClosed.pipe(
```

Comment:
There is another subscription for `this.overlayService.onClosed` event in method `_setupListeners()`. I suggest removing this subscription from here and moving `this.nativeElement.focus()` there. I know this code is only moved to a different location but I think it could be improved.

Reply:
@bkulov  I tried this, but focusing the this.nativeElement in all the cases will break the keyboard navigation and in the hgrid there is flickering. We do not need to focus native element in all the cases when overlay is closed, only in case when hide it from hideOverlays. The other way is to empty the overlayIDs in hideOverlays, but when the overlay is not closed by the hideOverlays, the array will not be correct. That is why I will leave it this way.